env/ups.go

```diff
@@ -0,0 +1,49 @@
+package env
+
+import (
+	"log"
+
+	"github.com/cloudfoundry-community/go-cfenv"
+)
+
+// WithUPSLookup configures the VarSet to use the Cloud Foundry user-provided
+// service with the given name as a lookup source.
+func WithUPSLookup(logger *log.Logger, app *cfenv.App, name string) VarSetOpt {
```

Comment:
Why do we have to pass a logger? (probably because it logs stuff)

Could we at least document that is is allowed to be nil to fail silently? (and maybe make it the last (as it is the least important) parameter?)

Reply:
So we're passing a logger because libraries shouldn't really log. Well, it's arguable, but in my experience, whenever a library logs it is annoying because you get these log messages that you can't stop. It's similar to panicking - something that you can't turn off but that you don't always want. I think the fact that there is a logger in this function is possibly an indication that it's _too_ domain-specific and maybe shouldn't even be in here. 

Alternatively, (per my note in the list at the start of this PR), maybe it should just return an error and let the caller decide if it wants to continue with a log message, or abort.